### PR TITLE
TASK: Some cleanup to the Flow persistence code

### PR DESCRIPTION
**Pull request by kdambekalns**

### CODE REVIEW

File: Neos.Flow/Classes/Persistence/Doctrine/Service.php [hunk 1]
```diff
@@ -696,7 +699,7 @@ protected function buildCodeFromSql(Configuration $configuration, array $sql): s
             array_unshift(
                 $code,
                 sprintf(
-                    '$this->abortIf($this->connection->getDatabasePlatform()->getName() !== %s, %s);',
+                    '$this->abortIf($this->connection->getDatabasePlatform()->getName() != %s, %s);',
```
Comment (albe) [inline Neos.Flow/Classes/Persistence/Doctrine/Service.php]: Is that change intentional? Because the variable given could also be `null`?

File: Neos.Flow/Classes/Persistence/Doctrine/Service.php [hunk 2]
```diff
@@ -641,27 +646,25 @@ class $className extends AbstractMigration
     /**
      * @return string
      */
-    public function getDescription(): string
+    public function getDescription()
```
Comment (albe) [inline Neos.Flow/Classes/Persistence/Doctrine/Service.php]: Return type removed? Was this breaking for some cases?

**Comment by albe** [comment]
ping @kdambekalns - would merge if my two questions are answered :)

File: Neos.Flow/Classes/Persistence/Doctrine/Service.php (hunk 2 above)
Comment (kdambekalns) [reply Neos.Flow/Classes/Persistence/Doctrine/Service.php]: It should match the signature from Doctrine… And in `doctrine/migrations` 1.8 that is without.

File: Neos.Flow/Classes/Persistence/Doctrine/Service.php (hunk 1 above)
Comment (kdambekalns) [reply Neos.Flow/Classes/Persistence/Doctrine/Service.php]: No, should be  `!==` as per https://github.com/doctrine/migrations/blob/1.8/lib/Doctrine/DBAL/Migrations/Tools/Console/Command/DiffCommand.php#L147

Comment (albe) [reply Neos.Flow/Classes/Persistence/Doctrine/Service.php]:
```suggestion
                    '$this->abortIf($this->connection->getDatabasePlatform()->getName() !== %s, %s);',
```

**Comment by kdambekalns** [comment]
Those changes are part of #1880 and/or #2231 as well…
